### PR TITLE
Part-I of Fix to 1081. Added a ThrowIfNotAvailable to ISolutionManager.

### PR DESCRIPTION
**Pull request by deepakaravindr**
Updated BuildIntegratedProjectUtility to handle cases where
solutionDirectory is null or is a relative path

Fixes https://github.com/NuGet/Home/issues/1081

@yishaigalatzer @emgarten

### CODE REVIEW

File: src/ProjectManagement/Strings.resx [hunk 1]
```diff
@@ -234,6 +234,9 @@
   <data name="RemovingPackageFromMSBuildProject" xml:space="preserve">
     <value>Removing package '{0}' from project '{1}'</value>
   </data>
+  <data name="SolutionDirectoryShouldBeANonNullFullPath" xml:space="preserve">
+    <value>Solution directory must be a non-null full path</value>
```
Comment (emgarten) [inline src/ProjectManagement/Strings.resx]: Is this shown to the user anywhere? non-null isn't a very user friendly term.

**Comment by emgarten** [comment]
:shipit:

File: src/ProjectManagement/Utility/BuildIntegratedProjectUtility.cs [hunk 2]
```diff
@@ -31,15 +31,22 @@ public static class BuildIntegratedProjectUtility
...
+            }
 
             var globalPackagesFolder = SettingsUtility.GetGlobalPackagesFolder(settings);
-            if (string.IsNullOrEmpty(solutionDirectory))
+            if (Path.IsPathRooted(globalPackagesFolder))
             {
                 return globalPackagesFolder;
             }
 
+            if (string.IsNullOrEmpty(solutionDirectory) || !Path.IsPathRooted(solutionDirectory))
```
Comment (yishaigalatzer) [inline src/ProjectManagement/Utility/BuildIntegratedProjectUtility.cs]: the message is inappropraite of the path is not rooted

File: src/ProjectManagement/Strings.resx (hunk 1 above)
Comment (yishaigalatzer) [inline src/ProjectManagement/Strings.resx]: This is never shown to the user in visual studio, this is an error of someone using the API. 

Check for null or empty, and provide an error saying it cannot be null

Then check for full path and provide an error saying the path must be rooted, and provide the path that was passed in.

File: src/PackageManagement/IDE/ISolutionManager.cs [hunk 3]
```diff
@@ -63,6 +71,8 @@ public interface ISolutionManager
         /// passed in.
         /// </returns>
         NuGetProject GetNuGetProject(string nuGetProjectSafeName);
+
+        void ThrowIfNotAvailable();
```
Comment (yishaigalatzer) [inline src/PackageManagement/IDE/ISolutionManager.cs]: remove from the interface

File: src/ProjectManagement/Strings.resx (hunk 1 above)
Comment (deepakaravindr) [inline src/ProjectManagement/Strings.resx]: Done. First checking for nullity. Throwing an ArgNull on solutionDirectory if null. If not, and if the path is not rooted, saying that the "SolutionDirectory '{0}' must be a rooted path."

File: src/ProjectManagement/Utility/BuildIntegratedProjectUtility.cs (hunk 2 above)
Comment (deepakaravindr) [inline src/ProjectManagement/Utility/BuildIntegratedProjectUtility.cs]: Fixed

File: src/PackageManagement/IDE/ISolutionManager.cs (hunk 3 above)
Comment (deepakaravindr) [inline src/PackageManagement/IDE/ISolutionManager.cs]: Done

**Comment by deepakaravindr** [comment]
Merged with https://github.com/NuGet/NuGet.PackageManagement/commit/7331e80954b61e377a0fb2b77722b75ac1efb2a9

**Pull request closed without merging**